Code Jam Wrap Up

janine9vn opened the pull request:
This updates the website now that the code jam is wrapped up.

It corrects wording and sidebars to show that the code jam is no longer currently on going. This happens primarily on the main page, the events page, and the code jam page.

Additionally this PR adds some information about the Code Jam winners with links to their projects.

This PR also goes a bit out scope and edits the contents on the main Events page, namely removing Hacktoberfest, updating the 2021 events calendar, and re-organizing the existing event blocks in an order that makes more sense.

janine9vn committed:
Remove code jam notice from front page

janine9vn committed:
Update wording on main Events page

janine9vn committed:
Update events featured
Re-ordered and updated the events listed on the main events page.
We're no longer participating in Hacktoberfest, so that was removed.
Game jams was moved to the bottom as we have not hosted one and will not
for the forseeable future.
PyWeek was moved above Advent of Code because AoC is one of the
last events we do in the year so it didn't make sense to have
PyWeek below it.

This also updated the Events Calendar on the sidebar to be accurate for
2021.

janine9vn committed:
Update Code Jam 8 as a finished event
This commit adds in the information about the winners of the code jam.

Additionally, it adds in Code Jam 8 as a prior event in the sidebar.

janine9vn committed:
Remove ongoing-codejam from cj page
Removes the sidebar block for ongoing code jams from the main code jam
events page.

Bluenix2 commented on pydis_site/templates/events/index.html:
```diff
@@ -9,58 +9,34 @@
-        <p>Each year, we organize a Winter Code Jam and a Summer Code Jam. During these events, members of our community will work together in teams to create something amazing using a technology we picked for them. One such technology that was picked for the Winter Code Jam 2020 was Kivy, a cross-platform GUI framework.</p>
-        <p>To help fuel the creative process, we provide a specific theme, like <strong>Ancient Technology</strong> or <strong>This App Hates You</strong>. At the end of the Code Jam, the projects are judged by Python Discord server staff members and guest judges from the larger Python community. The judges will consider creativity, code quality, teamwork, and adherence to the theme.</p>
+        <p>Each year, we organize at least one code jam, one during the summer and sometimes one during the winter. During these events, members of our community will work together in teams to create something amazing using a technology we picked for them. One such technology that was picked for the Summer 2021 Code Jam was text user interfaces (TUIS), where teams could pick from a pre-approved l ...
+        <p>To help fuel the creative process, we provide a specific theme, like <strong>Think Outside the Box</strong> or <strong>Early Internet</strong>. At the end of the Code Jam, the projects are judged by Python Discord server staff members and guest judges from the larger Python community. The judges will consider creativity, code quality, teamwork, and adherence to the theme.</p>
```
Why didn't these get cut off like the PyWeek ones?

janine9vn replied:
It's not cut off, it just doesn't have an icon to show, so the text takes the full width.
But this did make me realize the Advent of Code icon wasn't showing up, so I'm now fixed that.

Bluenix2 replied:
No I mean that PyWeek's lines you cut off at a certain length, while these ones go on for several hundreds of characters.

janine9vn replied:
I'm not sure what you're referencing. The length of the line is entirely dependent on if the word can fit or if it needs to go on the next line. Do you have a screenshot?

Bluenix2 replied:
Why doesn't these lines look like this?
![image](https://user-images.githubusercontent.com/38372706/133911944-6e696f5d-0f82-42d4-8360-7f96512223d5.png)

janine9vn replied:
Oh, you mean within the file itself, not how it displays. 

That's just what it was from the start. I only edited the contents of that page and the general order, not how the lines were broken up. I also tend to have word wrap turned on in my editor so it doesn't really bother me.

Bluenix2 commented on pydis_site/templates/events/sidebar/code-jams/previous-code-jams.html:
```diff
@@ -1,6 +1,7 @@
 <div class="box">
     <p class="menu-label">Previous Code Jams</p>
     <ul class="menu-list">
+        <li><a class="has-text-link" href="{% url "events:page" path="code-jams/8" %}">Code Jam 8: Think Outside the Box</a></li>
```
The theme was actually "Think Inside the Box".

janine9vn committed:
Use the correct theme and update image link
I now mention the correct theme which is "Think Inside the Box"

Additionally, this corrects the link to the icon for Advent of Code.
It was using an old link which is no longer correct.

fisher60 commented on pydis_site/templates/events/index.html:
```diff
@@ -9,58 +9,34 @@
-                <p>
-                    The <a class="has-text-link" href="{% url "events:page" path="game-jams/2020" %}">first edition of the Game Jam</a> ran from
-                    <strong>April 17, 2020 to April 26, 2020</strong>.
+                    For the past 15 years, <a href="https://pyweek.org">PyWeek</a> has been running a bi-annual game jam for the
```
This href should probably have `target="_blank"` and `rel="noopener"` the former for UX, the latter for security. Likewise with any other external links that may disrupt reading content on our site when the user may want to continue.

janine9vn committed:
Change external links to "_blank" and "noopener"
This changes external links to open in a new tab. It also adds
`rel="noopener"` for some bonus security and performance gains.

This commit only effects files that are within the scope of this PR.
It does not do this against other pages.

kosayoda commented on pydis_site/templates/events/pages/code-jams/8/_index.html:
```diff
@@ -20,10 +20,54 @@
+            <br/>
+            <a href="https://www.youtube.com/watch?v=WI9tgQeAfXw" title="Lovable Lobsters Demo Video" target="_blank" rel="noopener"><i class="fa fa-video"> </i> Demo video</a>
+            <br/>
+            <a href="https://github.com/A5rocks/code-jam-8" title="Lovable Lobsters GitHub Repository" target="_blank" rel="noopener"><i class="fa fa-github"></i> Githib Repository</a>
```
```suggestion
            <a href="https://github.com/A5rocks/code-jam-8" title="Lovable Lobsters GitHub Repository" target="_blank" rel="noopener"><i class="fa fa-github"></i> Github Repository</a>
```
Multiple instances of this throughout the page as far as I can tell~

janine9vn replied:
;-; ... You know, you'd think I'd be able to spell GitHub correctly at this point.

kosayoda commented on pydis_site/templates/events/pages/code-jams/8/_index.html:
```diff
@@ -20,10 +20,54 @@
+            <br/>
+        </p>
+
+        <h4><i class="fa fa-medal"></i> Robust Reindeer: Rubick's Cube</h4>
```
```suggestion
        <h4><i class="fa fa-medal"></i> Robust Reindeer: Rubik's Cube</h4>
```

fisher60 reviewed (approved):
All looks good to me.

kosayoda commented on pydis_site/templates/events/pages/code-jams/8/_index.html:
```diff
@@ -36,14 +80,14 @@ <h3 id="technology"><a href="#technology">Technology</a></h3>
-        For more information of TUIs and what's involved with such an interface, check out <a href="https://en.wikipedia.org/wiki/Text-based_user_interface">this wikipedia article</a>.
+        For more information of TUIs and what's involved with such an interface, check out <a href="https://en.wikipedia.org/wiki/Text-based_user_interface" target="_blank" rel="noopener">this wikipedia article</a>.
     </p>
     <h3 if="qualifier"><a href="#qualifier">The Qualifier</a></h3>
```
```suggestion
    <h3 id="qualifier"><a href="#qualifier">The Qualifier</a></h3>
```

kosayoda reviewed (changes requested):
Good work. I think the `Code Jam Winners` section could use a lot more whitespace, but that isn't a blocker for this PR.

janine9vn committed:
Correct typos and spacing
You would think I would be able to spell GitHub correctly at this point.

I've fixed the typos on this page and altered some of the spacing to
provide something nicer to glance over.

wookie184 reviewed (approved):
Good stuff

Bluenix2 commented on pydis_site/templates/events/sidebar/events-list.html:
```diff
@@ -1,10 +1,10 @@
 <div class="box">
```
What is this by the way?
![image](https://user-images.githubusercontent.com/38372706/134391486-8e3fd116-3e15-4769-9af3-d8a782d89775.png)

It is on the production site as well: https://pythondiscord.com/events/

whispersofthedawn replied:
It's the [breadcrumb](https://bulma.io/documentation/components/breadcrumb/) for the page.

Bluenix2 commented on pydis_site/templates/events/pages/code-jams/8/_index.html:
```diff
@@ -20,10 +20,58 @@
+
+            Just like a real world Rubik's cube, you can move this cube around to look at it from all sides. And, of course, you can rotate the individual discs it is made up of to first scramble up the order and then to try and solve it into perfect ordering again.
+        </p>
+        <p>
+            <a href="https://github.com/bjoseru/pdcj8-robust-reindeer" title="Robust Reindeer GitHub Repository" target="_blank" rel="noopener"><i class="fa fa-github"></i> GitHub Repository</a>
+            <br/>
```
Where's the video for this team?

janine9vn replied:
Purposely not included.

Bluenix2 reviewed (approved):
Now that my two questions have been resolved, looks good!

ChrisLovering committed:
Merge branch 'main' into cj-wrapup
